framework/src/modules/token/cc_method.ts

```diff
@@ -88,12 +85,16 @@ export class TokenInteroperableMethod extends BaseCCMethod {
 	public async beforeCrossChainMessageForwarding(ctx: CrossChainMessageContext): Promise<void> {
 		const { ccm } = ctx;
 		const methodContext = ctx.getMethodContext();
-		const messageFeeTokenID = await this._interopMethod.getMessageFeeTokenID(
+		const messageFeeTokenID = await this._interopMethod.getMessageFeeTokenIDFromCCM(
 			methodContext,
-			ccm.receivingChainID,
+			ccm,
 		);
 		const { ccmID } = getEncodedCCMAndID(ccm);
 
+		if (!messageFeeTokenID.equals(getMainchainTokenID(ctx.chainID))) {
```

Comment:
I think the call to `getMainchainTokenID` is wrong, since this function has no args according to its definition, so I guess it should be just 
```suggestion
		if (!messageFeeTokenID.equals(getMainchainTokenID())) {
```

Reply:
`getMainchainTokenID` (and soon `getTokenIDLSK`) would still take `chainID` arg in interoperability module. Only the function with the same name in token module takes no chainID arg.

Reply:
Updated